reflection.md

```diff
@@ -1,18 +1,66 @@
 # Reflection on implementing regular expressions of a DSL
 
 ## Which operators were easiest to implement and why?
+The Union `||` and Concatenation `~` operators were the easiest to implement. 
+These simply called existing case classes Union and Concat, so they were very
+straightforward.
 
 ## Which operators were most difficult to implement and why?
+For me, the repitition operator was difficult to implement because I was
+initially stumped by the `{}` curly braces and how to actually match curly braces
+before realizing it would have to be some special method. The implementation
+itself was not so difficult once realizing this.  The Star operator was also not
+immediately obvious, despite its simplicity, because the Scala syntax of 
+paramater-less functions are still relatively new to me. 
+
+I would also say the string2regex implementation was a bit difficult too. I
+had tried to implement it without explicitly specifying the types for things
+like `EPSILON` and `charList`, but ran into issues doing so. I also probably
+should have read the instructions and description of the regular language better
+as it took me a while to realize that the RegEx representation of a string is
+Concatenated `Literal`s.
 
 ## Comment on the design of this internal DSL
+   Nice things about the DSL: 	
+   
+   It is certainly very nice to use the DSL for longer RegEx conditions. The
+   implementation of the main example, `val pattern = "42" || ( ('a' <*>) ~ ('b' <+>) ~ ('c'{3}))` is much cleaner and intuitive using the DSL rather than the original implementation of 
+   ```
```

Comment:
A cleaner (and more common in mainstream languages) implementation is pretty hard to do with an internal DSL; usually regular expressions are parsed from strings.